airflow/models.py

```diff
@@ -1996,7 +1996,14 @@ def execute(self, context):
 
         Refer to get_template_context for more context.
         """
-        raise NotImplementedError()
+        if self.dag:
+            os.environ['AIRFLOW_DAG_ID'] = self.dag.dag_id
+        dagrun = context['dag_run']
+        if dagrun and dagrun.execution_date:
+                os.environ['AIRFLOW_DAGRUN_EXECUTION_DATE'] = dagrun.execution_date.isoformat()
+        os.environ['AIRFLOW_TASK_ID'] = self.task_id
+        if self.start_date:
+            os.environ['AIRFLOW_TASK_INSTANCE_EXECUTION_DATE'] = self.start_date.isoformat()
```

Comment:
It'd be nice to have a 100% predictable pattern where you replace dots by double underscore.

dag.dag_id becomes `AIRFLOW_CTX__DAG__DAG_ID`

I'd also go with a longer prefix `AIRFLOW_CTX__` to avoid potential collisions.